app/src/main/java/com/kodeco/android/coordplot/MainActivity.kt

```diff
@@ -81,17 +88,30 @@ fun PlotSurfacePreview() {
 
 @Composable
 fun Map(xPercent: Float, yPercent: Float, modifier: Modifier = Modifier) {
+    val density = LocalDensity.current.density
+    var boxSize by remember { mutableStateOf(Size(0, 0)) }
+
     Box(
         modifier = modifier
             .size(PLOT_SIZE.dp)
+            .aspectRatio(1f) // Ensure the plot is a square (1:1 aspect ratio)
             .background(PLOT_BACKGROUND_COLOR)
+            .onGloballyPositioned { coordinates ->
+                boxSize = Size(
+                    ((coordinates.size.width / density).toInt()),
+                    ((coordinates.size.height / density).toInt())
+                )
+            }
```

Comment:
@zlippard Is this something that's commonly used? I tried to use: 
```kotlin 
.onSizeChanged { size ->
    boxSize = Size(size.width, size.height)
}
```
but it was out of proportion - so is that "Density" thing a common practice?

Reply:
Good question! So managing density can be important at times however it's not usually a common practice (at least not a common practice to go from dp to pixels as you are having to do with `onGloballyPositioned`

So, both `onSizeChanged` and `onGloballyPositioned` are calls that are invoked during the [Layout](https://developer.android.com/jetpack/compose/phases) phase of composition.   While it works, this might cause extra work and incur _extra_ recomposition loops during the rendering of the UI. For this case, we can avoid the need for using either of these functions.  I'll explain in other comments the other changes we need, but suffice it say, we should be able to remove `onGloballyPositioned` here in your code.

Reply:
Hmm @zlippard Ok so the reason I did these changes (see original commit [825758c0eed33ea52]) was because I wanted the plot to "always fit" irrelevant of the `PLOT_SIZE` value. While `300` may work, if someone sets it to say `700` then it doesn't fit on a Pixel 7 Pro and the x&y coords then break: 


https://github.com/heidiproske/kodeco_android_bootcamp_coordplot/assets/6706831/21c16fe0-4a0d-47a9-aca0-001628a4e0f4


So what's why I made the changes that you're suggesting I revert here 😁 (IF I revert them - then I'm back to that bug ☝️) So I'll keep them in for now cuz I'd really like to understand how to recalculate the sizes "once" on a screen draw to ensure that they always fit! Will join the next office hours to ask.

Reply:
So sorry I missed this replies @heidiproske! I see what you were trying to do now though and that makes total sense. I understand now that you were trying to consider a more dynamic sizing of the box.  If you're able to make it to the office hours this week let's work through this problem.  It's likely that what you had originally with the `onGloballyPositioned` is indeed what you want but we might be able to get around the need for dividing by the density and converting to/from pixels and DP.  

I likely misunderstood what the code was attempting to do here.  That is my mistake!